### PR TITLE
(#1015) Path exists check retains path scope specified by user, adds semi-colon to end of path

### PR DESCRIPTION
**Pull request by DarwinJS**
(GH 1015) Path exists check retains path scope specified by user,  adds semi-colon to end of path

Also path exists check finds path whether or not trailing slash matches requested path, added alias 'path' for 'pathtoinstall' to align with powershell parameter naming, used chocolatey method Update-SessionEnvironment instead of simple path concetanation.

#1015

### CODE REVIEW

**Comment by DarwinJS** [comment]
Looking for feedback on the following possible additional changes:

- Thinking I should also change "write-host" to best practice "write-output" - however, I'm not sure if that has implications due to the fact chocolatey hosts powershell by default?
- New scope option 'All' for adding path to both User and Machine
- Check if path is in other scope and log that fact
- Give a warning that paths added to User scope will ONLY be available to that user and not the System, services nor other users.

File: src/chocolatey.resources/helpers/functions/Install-ChocolateyPath.ps1 [hunk 1]
```diff
@@ -65,28 +65,23 @@ Set-EnvironmentVariable
...
   $originalPathToInstall = $pathToInstall
-
-  #get the PATH variable
-  Update-SessionEnvironment
-  $envPath = $env:PATH
-  if (!$envPath.ToLower().Contains($pathToInstall.ToLower()))
+  #array drops blanks (one of which is always created by final semi-colon)
+  $actualPathArray = (Get-EnvironmentVariable -Name 'Path' -Scope $pathType -PreserveVariables).split(';',[System.StringSplitOptions]::RemoveEmptyEntries)
+  #checks for match with and without trailing slash
+  if (($actualpathArray -inotcontains $pathToInstall.ToLower()) -AND ($actualpathArray -inotcontains "$(($pathToInstall + '\').ToLower())"))
```
Comment (Litee) [inline src/chocolatey.resources/helpers/functions/Install-ChocolateyPath.ps1]: $actualpathArray is compared with lower case version of $pathToInstall. Where $actualpathArray is converted to lower case?

Comment (DarwinJS) [inline src/chocolatey.resources/helpers/functions/Install-ChocolateyPath.ps1]: The "i" in "-inotcontains" causes the entire comparison operation to be "case **I**nsensitive".  Technically this make "ToLower()" redundant.

**Comment by dragon788** [comment]
@DarwinJS I think from talking with @ferventcoder [about this on Gitter](https://gitter.im/chocolatey/choco?at=581bb24131c5cbef43b21447) Write-Host is best because it avoids major pitfalls of Write-Output's behavior, if you have Write-Output inside a function it automatically forces itself into the return value. There is a new Write-Information, but that is a PS5 only function iirc.

**Comment by DarwinJS** [comment]
Trying to understand why this isn't getting merged?

**Comment by ferventcoder** [comment]
Apologies @DarwinJS - this change wasn't under vNext, so it didn't get in for 0.10.4 - 0.10.5 was a bug fix release for a gating issue in 0.10.4. 0.10.6 is fixing an issue that came in from a change to how xml files are saved. 

It will make it in, and I did take a look and notice it may change the current behavior of looking at the entire path (breaking change), even if that is technically the right thing to do, and realized this changes a lot of code so it requires more scrutiny. Since it wasn't technically scheduled for the next release, it did not get merged. If we can agree that this is more of an annoyance than a critical fix - it may add an item to path twice, but it won't necessarily stop choco from working. HTH

**Comment by DarwinJS** [comment]
@ferventcoder - no need to apologize - I am just trying to understand.
Is the same true of https://github.com/chocolatey/choco/pull/1019 ?

**Comment by jedfrechette** [comment]
Has there been any progress on getting this merged?

I just ran in to the issue described in #1015, where the Machine PATH was silently not getting updated because the path I was trying to add already existed on the User PATH.  I never would have guessed what the problem was without finding that bug.

**Comment by vexx32** [comment]
Thanks for the contribution here, and apologies that this was not reviewed for a long while.

This has been resolved during the rewrite of this command to C# which shipped in 2.3.0 and the issue this seeks to resolve is no longer present; see here: https://github.com/chocolatey/choco/issues/1015#issuecomment-2359180577

As such, I'll be closing this PR, but thank you for the contribution nonetheless! 💜

**Pull request closed without merging**